rules/hmap/hmap.c

```diff
@@ -191,14 +191,22 @@ int hmap_save(HeaderMap* hmap, char* path) {
         perror(path);
         return 1;
     }
-    int rc = 0;
```

Comment:
@ob this change removed the `close` on failure which I previously had handled.  I'm not really sure it matters in the context of this program but it's probably more correct to close the open descriptors in general

Reply:
it'll close on exit but sure, probably cleaner to close it....